src/System.Private.CoreLib/src/System/Decimal.DecCalc.cs

```diff
@@ -301,7 +302,7 @@ private static uint Div96By32(uint[] rgulNum, uint ulDen)
              *   None.
              *
              ***********************************************************************/
-            private static uint Div96By64(System.Collections.Generic.IList<uint> rgulNum, Split64 sdlDen)
+            private static uint Div96By64(uint* rgulNum, Split64 sdlDen)
```

Comment:
These unsafe casts in number of places look dangerous. I am wondering whether it would be possible to wrap the pointer in a struct and verify that there are no out-of-bound accesses using asserts in debug builds at least.

Reply:
I changed these back to pointers like in the [C++ code](https://github.com/dotnet/coreclr/blob/master/src/classlibnative/bcltype/decimal.cpp#L933) to avoid allocations. I'll add some `Debug.Assert`s in places were it's not obvious that the pointer access is safe.

Reply:
I managed to get rid of pointers almost everywhere, except in a few loops in ADD/MUL where I added `Debug.Assert` guards. Perf is about the same. DIV related changes are not yet committed.